ducktape/tests/reporter.py

```diff
@@ -304,3 +307,41 @@ def format_report(self):
 
     def report(self):
         self.format_report()
+
+
+class TestSuiteGeneratorReporter(SummaryReporter):
+
+    def __init__(self, results):
+        super().__init__(results)
+        self.separator = ''.join(["=" * self.width])
+
+    def format_line(self, result):
+        line = f'{result.file_name}::{result.cls_name}.{result.function_name}'
+        if result.injected_args:
+            injected_args_str = json.dumps(result.injected_args, separators=(',', ':'))
+            line += f'@{injected_args_str}'
```

Comment:
do we want to make some special checks here to make sure the args are serializable to json?

Reply:
good idea

Reply:
Huh, actually it has to be serializable, otherwise runner_client will raise